src/names.c

```diff
@@ -1611,6 +1611,9 @@ static void hint_for_typo(scope_t *top_scope, diag_t *d, ident_t name,
 
 static type_t get_result_type(nametab_t *tab, tree_t decl)
 {
+   if (!tree_has_type(decl))
+      return type_new(T_NONE);
+
```

Comment:
I think this works too

```suggestion
   if (tree_kind(decl) == T_ALIAS && !tree_has_type(decl))
      return tree_type(tree_value(decl));

```

It makes a bit more sense when reading it ("if the thing is an alias declaration and it didn't have a type indication then use the type of aliased object (which will be `T_NONE` here").

Reply:
Agreed, that conveys more meaning and also fixes the crash.